universal-application-tool-0.0.1/app/auth/UatProfile.java

```diff
@@ -0,0 +1,62 @@
+package auth;
+
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+
+import com.google.common.base.Preconditions;
+import java.time.Clock;
+import java.util.Comparator;
+import java.util.concurrent.CompletableFuture;
+import models.Account;
+import models.Applicant;
+import org.pac4j.core.profile.CommonProfile;
+
+public class UatProfile extends CommonProfile {
+  private final Clock clock;
+
+  public UatProfile() {
+    this(Clock.systemDefaultZone());
+  }
+
+  public UatProfile(Clock clock) {
+    super();
+    this.clock = clock;
+  }
+
+  public CompletableFuture<Applicant> getApplicant() {
+    return this.getAccount()
+        .thenApply(
+            (a) ->
+                a.getApplicants().stream()
+                    .sorted(
+                        Comparator.comparing(
+                            (applicant) -> applicant.getApplicantData().getCreatedTime()))
+                    .findFirst()
+                    .orElseThrow());
```

Comment:
Suggestion (blocking): to be on the safe side this should explicitly specify the `HttpExecutionContext`. Inject it, then

```java
.thenApply(() -> {...}, httpExecutionContext.current());
```